plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/protobuf/TestKafkaProtobufWithSchemaRegistryMinimalFunctionality.java

```diff
@@ -217,20 +221,48 @@ public void testSchemaWithImportDataTypes()
-                                    AS ROW(
-                                        string_column VARCHAR,
-                                        integer_column INTEGER,
-                                        long_column BIGINT,
-                                        double_column DOUBLE,
-                                        float_column REAL,
-                                        boolean_column BOOLEAN,
-                                        number_column VARCHAR,
-                                        timestamp_column TIMESTAMP(6),
-                                        bytes_column VARBINARY)))""");
+                        VALUES (
+                            ARRAY[CAST('Search' AS VARCHAR)],
+                            MAP(CAST(ARRAY['Key1'] AS ARRAY(VARCHAR)), CAST(ARRAY['Value1'] AS ARRAY(VARCHAR))),
+                            CAST(ROW('Trino', 1, 493857959588286460, 3.14159265358979323846, 3.14, True, 'ONE', TIMESTAMP '2020-12-12 15:35:45.923', to_utf8('Trino'))
+                                AS ROW(
+                                    string_column VARCHAR,
+                                    integer_column INTEGER,
+                                    long_column BIGINT,
+                                    double_column DOUBLE,
+                                    float_column REAL,
+                                    boolean_column BOOLEAN,
+                                    number_column VARCHAR,
+                                    timestamp_column TIMESTAMP(6),
+                                    bytes_column VARBINARY)))""");
+    }
+
+    @Test
+    public void testOneOf()
+            throws Exception
+    {
+        String topic = "topic-schema-with-oneof";
+        assertNotExists(topic);
+
+        String stringData = "stringColumnValue1";
+
+        ProtobufSchema schema = (ProtobufSchema) new ProtobufSchemaProvider().parseSchema(Resources.toString(getResource("protobuf/test_oneof.proto"), UTF_8), List.of(), true).get();
+
+        Descriptor descriptor = schema.toDescriptor();
+        DynamicMessage message = DynamicMessage.newBuilder(descriptor)
+                .setField(descriptor.findFieldByName("stringColumn"), stringData)
+                .build();
+
+        ImmutableList.Builder<ProducerRecord<DynamicMessage, DynamicMessage>> producerRecordBuilder = ImmutableList.builder();
+        producerRecordBuilder.add(new ProducerRecord<>(topic, createKeySchema(0, getKeySchema()), message));
+        List<ProducerRecord<DynamicMessage, DynamicMessage>> messages = producerRecordBuilder.build();
+        testingKafka.sendMessages(messages.stream(), producerProperties());
+        waitUntilTableExists(topic);
+
+        assertThat(query(format("SELECT testOneOfColumn FROM %s", toDoubleQuoted(topic))))
+                .matches("""
+                        VALUES (JSON '{"stringColumn":"%s","integerColumn":null,"longColumn":null,"doubleColumn":null,"floatColumn":null,"booleanColumn":null}')
```

Comment:
Do we want all possible columns in the `oneof` to be in the JSON string, leaving all other values as `null`? Or just a single column e.g. `{"stringColumn":"value"}`?

Reply:
I think having a single field populated would be better because otherwise you can't differentiate that `stringColumn=null` because stringColumn was set to `null` or because some other field existed and stringColumn didn't exist.

Reply:
Makes sense and I agree. I'll update to only write a field if it is present. Protobuf doesn't do normal `null` values, but I have added a test case for an empty message (which results in an empty JSON value `{}`).